openssl/src/dsa.rs

```diff
@@ -153,6 +177,31 @@ impl Dsa<Private> {
             Ok(dsa)
         }
     }
+
+    /// Create a DSA key pair with the given parameters
+    ///
+    /// `p`, `q` and `g` are the common parameters.
+    /// `priv_key` is the private component of the key pair.
+    /// The corresponding public component is calculated from the private component.
+    pub fn from_private_components(
+        p: BigNum,
+        q: BigNum,
+        g: BigNum,
+        priv_key: BigNum,
+    ) -> Result<Dsa<Private>, ErrorStack> {
```

Comment:
I'd probably keep this as purely calling into the various setters rather than computing the public key itself.

Reply:
Done. Should there be any checks for consistent keys? I'm not sure what openssl does when private and public key are not matching. Looking at the RSA code, there no checks either, so I guess it is fine?

Reply:
Thanks! We don't generally provide any extra checking on top of what OpenSSL already does, except for cases where that can cause memory safety issues. That's why we parameterize these types over what information they contain, for example.